src/node/utxo_snapshot.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/utxo_snapshot.h>
+
+#include <fs.h>
+#include <logging.h>
+#include <streams.h>
+#include <uint256.h>
+#include <validation.h>
+
+#include <cstdio>
+#include <optional>
+
+namespace node {
+
+bool WriteSnapshotBaseBlockhash(CChainState& snapshot_chainstate)
+{
+    AssertLockHeld(::cs_main);
+    assert(snapshot_chainstate.m_from_snapshot_blockhash);
+
+    const std::optional<fs::path> chaindir = snapshot_chainstate.CoinsDB().StoragePath();
+    assert(chaindir); // Sanity check that chainstate isn't in-memory.
+    const fs::path write_to = *chaindir / node::SNAPSHOT_BLOCKHASH_FILENAME;
```

Comment:
question in https://github.com/bitcoin/bitcoin/commit/d59db4ff4b585d2b86b68ea0db206ec1c0cc5022: why can't the blockhash be read from the metadata in the dump?

Reply:
By "dump" do you mean the UTXO snapshot? If so, I don't think we can expect users to keep the snapshot on-hand after they've loaded it; it's essentially unnecessary at that point, and multiple GBs.